src/download/mod.rs

```diff
@@ -469,18 +469,17 @@ mod curl {
                 // Listen for headers and parse out a `Content-Length` (case-insensitive) if it
                 // comes so we know how much we're downloading.
                 transfer.header_function(|header| {
-                    if let Ok(data) = str::from_utf8(header) {
-                        let prefix = "content-length: ";
-                        if data.to_ascii_lowercase().starts_with(prefix) {
-                            if let Ok(s) = data[prefix.len()..].trim().parse::<u64>() {
-                                let msg = Event::DownloadContentLengthReceived(s + resume_from);
-                                match callback(msg) {
-                                    Ok(()) => (),
-                                    Err(e) => {
-                                        *cberr.borrow_mut() = Some(e);
-                                        return false;
-                                    }
-                                }
+                    let prefix = "content-length: ";
+                    if let Ok(data) = str::from_utf8(header)
```

Comment:
This is cool and stuff but I feel like it would be more readable as a series of early returns with let-else.

Reply:
I think the new version (modulo the "ascii case insensitive prefix match" optimization") is not very clear because it repeats the default return value `true` so many times. It's quite unfortunate that we can't use `let-chains` with `let-else` yet...

Reply:
It occurs to me that this `curl`-related code will be deleted sooner or later anyway, so I'm not going to cause much churn here blocking on this tiny detail :)